File: 2021/day2.py

data = []
with open("day2-input.txt") as f:
    data = [line.strip() for line in f.readlines()]


def part1():
  x = 0
  y = 0
  for el in data:
    command = el.split(" ")
    dir = command[0]
    amount = int(command[1])

    if dir == 'forward':
      x += amount
    elif dir == 'up':
      y -= amount
    else:
      y += amount
  
  return x * y


def part2():
  x = 0
  y = 0
  aim = 0

  for el in data:
    command = el.split(" ")
    dir = command[0]
    amount = int(command[1])

    if dir == 'down':
      aim += amount
    elif dir == 'up':
      aim -= amount
    else:
      x += amount
      y += aim*amount
  
  return x * y

print("Part I:", part1())
print("Part II:", part2())

File: 2021/kotlin/day2.kt
import java.io.File

fun main() {
  fun getAllLinesFromFile(filename: String): List<String> =
    File(filename).readLines().map { it.trim() }

  fun part1(input: List<String>): Int {
    var x: Int = 0;
    var y: Int = 0;

    input.forEach { el ->
      val command = el.split(" ")
      val dir = command[0]
      val amount = command[1].toInt()

      if (dir == "forward") x += amount
      else if (dir == "up") y-= amount
      else y += amount
    }

    return x * y
  }

  fun part2(input: List<String>): Int {
    var x: Int = 0
    var y: Int = 0
    var aim: Int = 0

    input.forEach { el ->
      val command = el.split(" ")
      val dir = command[0]
      val amount = command[1].toInt()

      if (dir == "down") aim += amount
      else if (dir == "up") aim -= amount
      else {
        x += amount
        y += aim*amount
      }
    }

    return x * y
  }

  val input = getAllLinesFromFile("../data/day2-input.txt")
  println("Part I: " + part1(input).toString())
  println("Part II: " + part2(input).toString())

  
}
